staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types_jsonschema.go

```diff
@@ -196,6 +196,15 @@ type ValidationRule struct {
 	// If unset, the message is "failed rule: {Rule}".
 	// e.g. "must be a URL with the host matching spec.host"
 	Message string
+	// MessageExpression declares a CEL expression that evaluates to the validation failure message that is returned when this rule fails.
+	// Since MessageExpression is used as a failure message, it must evaluate to a string at compilation time.
+	// If both Message and MessageExpression are present on a Rule, then MessageExpression will be used if validation
+	// fails. If MessageExpression results in a runtime error, the runtime error is logged, and the validation failure message is produced
+	// as if the MessageExpression field were unset.
```

Comment:
nit: Make it super clear that the message CEL expression has access to all the same variables as the `rule` expression and that the only difference is the return type?